recipe/conda_forge_ci_setup/upload_or_check_non_existence.py

```diff
@@ -158,7 +158,7 @@ def upload_or_check(
     allowed_dist_names = get_built_distribution_names(recipe_dir, variant)
```

Comment:
I think this pulls all of the names that are built. So we can look through every subdir and make sure it is in this list.

Reply:
This check is already there: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/118/files#diff-2c55b5e34665621d4e425217f0f28912R192

So i'd just grab everything and prune it later.

Reply:
It gives use the names, but not the subdirs that they are located in

Reply:
sure but that's fine - any package with the right name can be uploaded regardless of subdir

Reply:
also the recipes are fully rendered in that function, so the correct subdir could be returned too.